components/function-controller/pkg/apis/serverless/v1alpha1/function_validation.go

```diff
@@ -0,0 +1,123 @@
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/validation"
+	v1validation "k8s.io/apimachinery/pkg/apis/meta/v1/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"knative.dev/pkg/apis"
+)
+
+var reservedEnvs = []string{
+	"K_REVISION",
+	"K_CONFIGURATION",
+	"K_SERVICE",
+	"FUNC_RUNTIME",
+	"FUNC_HANDLER",
+	"FUNC_TIMEOUT",
+	"FUNC_PORT",
+	"PORT",
+	"MOD_NAME",
+	"NODE_PATH",
+}
+
+func (fn *Function) Validate(_ context.Context) (errors *apis.FieldError) {
+	errors = fn.validateObjectMeta(errors, "metadata")
+
+	spec := fn.Spec
+	errors = spec.validateSource(errors, "spec.source")
+	errors = spec.validateDeps(errors, "spec.deps")
+	errors = spec.validateEnv(errors, "spec.env")
+	errors = spec.validateLabels(errors, "spec.labels")
+	errors = spec.validateReplicas(errors, "spec")
```

Comment:
This is only the prefix for function. When an error occurs path will be set to `spec.minReplicas` or `spec.maxReplicas`. It's because we don't have `replicas` field